src/rules/noParameterPropertiesRule.ts

```diff
@@ -40,20 +41,21 @@ export class Rule extends Lint.Rules.AbstractRule {
     }
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithWalker(new NoParameterPropertiesWalker(sourceFile, this.getOptions()));
+        return this.applyWithFunction(sourceFile, walk);
     }
 }
 
-export class NoParameterPropertiesWalker extends Lint.RuleWalker {
-    public visitConstructorDeclaration(node: ts.ConstructorDeclaration) {
-        const parameters = node.parameters;
-        for (const parameter of parameters) {
-            if (parameter.modifiers != null && parameter.modifiers.length > 0) {
-                const errorMessage = Rule.FAILURE_STRING_FACTORY((parameter.name as ts.Identifier).text);
-                const lastModifier = parameter.modifiers[parameter.modifiers.length - 1];
-                this.addFailureFromStartToEnd(parameter.getStart(), lastModifier.getEnd(), errorMessage);
+function walk(ctx: Lint.WalkContext<void>) {
+    return ts.forEachChild(ctx.sourceFile, function cb(node): void {
+        if (node.kind === ts.SyntaxKind.Constructor) {
+            for (const parameter of (node as ts.ConstructorDeclaration).parameters) {
+                if (isParameterProperty(parameter)) {
+                    ctx.addFailure(parameter.getStart(ctx.sourceFile),
+                                   parameter.name.pos,
+                                   Rule.FAILURE_STRING_FACTORY((parameter.name as ts.Identifier).text));
```

Comment:
it wouldn't crash, it would just result in `undefined`. Anyways, I fixed it and added a test to ensure it works